spark/src/main/scala/geotrellis/spark/io/cassandra/CassandraDriver.scala

```diff
@@ -0,0 +1,54 @@
+package geotrellis.spark.io.cassandra
+
+import java.nio.ByteBuffer;
+
+import geotrellis.spark._
+import geotrellis.raster._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import com.datastax.spark.connector.rdd.CassandraRDD
+import com.datastax.spark.connector._
+
+import com.datastax.driver.core.DataType.{text, blob}
+import com.datastax.driver.core.schemabuilder.SchemaBuilder
+import com.datastax.driver.core.Session
+
+class TableNotFoundError(table: String) extends Exception(s"Target Cassandra table '$table' does not exist.")
+
+trait CassandraDriver[K] extends Serializable {
+  def applyFilter(rdd: CassandraRDD[(String, ByteBuffer)], layerId: LayerId, filterSet: FilterSet[K]): RDD[(String, ByteBuffer)]
+  def decode(rdd: RDD[(String, ByteBuffer)], metaData: RasterMetaData): RasterRDD[K]
+  def rowId(id: LayerId, key: K): String
+
+  def load(sc: SparkContext, keyspace: String)(
+    id: LayerId, metaData: RasterMetaData, table: String, filters: FilterSet[K]): RasterRDD[K] = {
+
+    val rdd: CassandraRDD[(String, ByteBuffer)] = 
+      sc.cassandraTable[(String, ByteBuffer)](keyspace, table).select("id", "tile")
+
+    val filteredRDD = applyFilter(rdd, id, filters)
```

Comment:
This is the crux of why building on of these catalogs is hard...

The current code as you have it loads the whole RDD, and then applies a filter. That means we load *every* tile. Unless you are working with the whole RDD, you are not going to want to pay the price of loading each element into memory...that loads each individual tile into memory at some point. Very slow.

So, we need to find a way to filter things on the Cassandra side. In the Accumulo catalog, we send along Accumulo filter information so that the tiles are filtered by Accumulo before being returned. That way we don't load every tile, only the tiles we want.

So, the question is, how can we only load what we want out of Cassandra?